### PR TITLE
Feat add support descriptors match case

### PR DESCRIPTION
**Pull request by FabienCeppic**

### CODE REVIEW

File: src/entries.ts [hunk 1]
```diff
@@ -1,2 +1,5 @@
 export const entries = (obj: Record<string, unknown>): ReadonlyArray<[string, unknown]> => 
   Object.keys(obj).map(key => [key, obj[key]])
+
+export const fromEntries = (entries: ReadonlyArray<[string, unknown]>): Record<string, unknown> =>
```
Comment (Methrat0n) [inline src/entries.ts:4]: Remove it please, its not used anymore.

**Review by Methrat0n — CHANGES_REQUESTED** [review]

File: src/does-match.ts [hunk 2]
```diff
@@ -1,26 +1,36 @@
...
 import { includes } from "./includes"
+import { Descriptor } from "./descriptors"
 
-export const doesMatch = (target: object, value: object): boolean => {
+const selectByKeys = (keys: ReadonlyArray<string>, obj: Record<string, unknown>): Record<string, unknown> =>
+  entries(obj).reduce((acc, [k, v]) => includes(keys, k) ? { ...acc, [k]: v} : acc, {})
 
+export const doesMatch = (target: Record<string, Descriptor<unknown>> | object, value: object): boolean => {
+  
+  if (entries(target as Record<string, unknown>).length === 0)
```
[the remark this answers is not in the document]
Comment (FabienCeppic) [reply src/does-match.ts]: Ok

File: src/does-match.ts [hunk 3]
```diff
@@ -1,26 +1,36 @@
...
+    return true
+
+  const targetWithoutDescriptor: Array<[string, unknown]> = []
```
[the remark this answers is not in the document]
Comment (FabienCeppic) [reply src/does-match.ts]: Ok

File: src/does-match.ts [hunk 4]
```diff
@@ -1,26 +1,36 @@
...
+  
+  entries(target as Record<string, unknown>).map((entry) => {
```
[the remark this answers is not in the document]
Comment (FabienCeppic) [reply src/does-match.ts]: Ok

**Pull request merged**